### PR TITLE
CI: Fixes to conda environment names

### LINKED ISSUE #23851: CLN: Fixes to conda environment names
We have some inconsistency regarding the names of the conda environment. At the moment, all environments for the CI are named `pandas`, and the environment created locally following the set up instructions (and using `environment.yml`) is named `pandas-dev`.

Then, inside several scripts (e.g. `ci/code_checks.sh`) we do a `source activate pandas`. Which means that the script works well in CI jobs, but it fails locally unless we also have a `pandas` environment locally.

Another thing is that in the CI config we use a `CONDA_ENV` environment variable, and we ignore the conda environment names in the yaml files.

What I would do is:
- Use `pandas-dev` as the environment name in all our conda yaml files
- Create the environments with the environment name in the used yaml file (do not override with `-n $CONDA_ENV` as we do now)
- Do not use `source activate` inside the scripts (use it in the steps in the CI config files when needed). This way when running scripts locally, the current environment will be used

@jreback @TomAugspurger @jorisvandenbossche any objection?

### PR DESCRIPTION
**Pull request by charlesdong1991**
- [x] closes #23851 
- [ ] tests added / passed
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

**Comment by datapythonista** [comment]
@charlesdong1991 are you still working on it? The rest of the changes in the scripts and the travis and azure config described in the issue are missing.

**Comment by charlesdong1991** [comment]
Hi, so sorry that I am having dinner out, but I will carry on working after
back home.

**Comment by datapythonista** [comment]
Not a problem at all, just said, as it make things more efficient. Enjoy your dinner, and thanks for working on this.

File: ci/incremental/setup_conda_environment.cmd [hunk 1]
```diff
@@ -11,11 +11,10 @@ call deactivate
 @rem Display root environment (for debugging)
 conda list
 @rem Clean up any left-over from a previous build
-conda remove --all -q -y -n %CONDA_ENV%
+
```
Comment (datapythonista) [inline ci/incremental/setup_conda_environment.cmd]: we may need this, but you can replace the `%CONDA_ENV%` by `pandas-dev`

File: ci/incremental/setup_conda_environment.sh [hunk 2]
```diff
@@ -12,15 +12,15 @@ conda list
 # Clean up any left-over from a previous build
 # (note workaround for https://github.com/conda/conda/issues/2679:
 #  `conda env remove` issue)
-conda remove --all -q -y -n $CONDA_ENV
+conda remove --all -q -y
```
Comment (datapythonista) [inline ci/incremental/setup_conda_environment.sh]: can you leave the `-n pandas-dev`

File: ci/incremental/setup_conda_environment.sh [hunk 3]
```diff
@@ -12,15 +12,15 @@ conda list
...
 
 echo
 echo "[create env]"
-time conda env create -q -n "${CONDA_ENV}" --file="${ENV_FILE}" || exit 1
+time conda env create -q --file="${ENV_FILE}" || exit 1
 
 # Activate first
 set +v
-source activate $CONDA_ENV
+source activate pandas-dev
```
Comment (datapythonista) [inline ci/incremental/setup_conda_environment.sh]: this one can be removed, also the `set +/- v` around it

**Review by datapythonista — CHANGES_REQUESTED** [review]
Looks good, thanks for working on this @charlesdong1991. Couple of comments, and they should fix the CI which is in red. If after changing them, you see the CI still red, you may want to check what else is needed.

**Comment by charlesdong1991** [comment]
@datapythonista sorry, this is very wrong... i think i did something very bad when catching up with the latest change, i think i need to re-create a pr... sorry

**Pull request closed without merging**